applications/CoSimulationApplication/custom_utilities/conversion_utilities.h

```diff
@@ -59,10 +59,18 @@ class KRATOS_API(CO_SIMULATION_APPLICATION) ConversionUtilities
     ///@name Operations
     ///@{
 
+    template<class TDataType>
     static void ConvertElementalDataToNodalData(
         ModelPart& rModelPart,
-        const Variable<array_1d<double,3> >& rElementalVariable,
-        const Variable<array_1d<double,3> >& rNodalVariable );
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
+
+    template<class TDataType>
+    static void ConvertNodalDataToElementalData(
+        ModelPart& rModelPart,
+        const Variable<TDataType>& rElementalVariable,
+        const Variable<TDataType>& rNodalVariable );
```

Comment:
These definitely need some documentation. It's not clear to me at all that `ConvertNodalDataToElementalData` basically sums up nodes' values in each element.